truss/templates/server/model_wrapper.py

```diff
@@ -247,29 +236,58 @@ def from_model(cls, model) -> "ModelDescriptor":
         else:
             return_annotation = inspect.signature(model.predict).return_annotation
 
-        if hasattr(model, "setup_environment"):
-            setup_environment = MethodDescriptor.from_method(
-                model.setup_environment, "setup_environment"
+        return TrussSchema.from_signature(parameters, return_annotation)
+
+    @classmethod
+    def _safe_extract_descriptor(
+        cls, model: Any, method: str
+    ) -> Union[MethodDescriptor, None]:
+        if hasattr(model, method):
+            return MethodDescriptor.from_method(getattr(model, method), method)
+        return None
+
+    @classmethod
+    def from_model(cls, model) -> "ModelDescriptor":
```

Comment:
Changes to this method are intended to be no-ops _except_ for the addition of completions / chats on L271/272. Everything else is a refactor